src/main/java/gregtech/api/metatileentity/MetaTileEntity.java

```diff
@@ -425,12 +426,25 @@ public final void onCoverLeftClick(EntityPlayer playerIn, CuboidRayTraceResult r
      * @return true if something happened, so animation will be played
      */
     public boolean onRightClick(EntityPlayer playerIn, EnumHand hand, EnumFacing facing, CuboidRayTraceResult hitResult) {
+        ItemStack heldStack = playerIn.getHeldItem(hand);
         if (!playerIn.isSneaking() && openGUIOnRightClick()) {
             if (getWorld() != null && !getWorld().isRemote) {
                 MetaTileEntityUIFactory.INSTANCE.openUI(getHolder(), (EntityPlayerMP) playerIn);
             }
             return true;
         } else {
+            // Attempt to rename the MTE first
+            if (heldStack.getItem() == Items.NAME_TAG) {
+                if (playerIn.isSneaking() && heldStack.getTagCompound() != null && heldStack.getTagCompound().hasKey("display")) {
```

Comment:
Optional: Having these keys to NBT as const in some global file would be beneficial to find all places in which we are doing something with it. In case we would need to return to it later.